pkg/app/run.go

```diff
@@ -16,140 +16,200 @@ package app
+			Name:  "detach",
+			Usage: "Detach from the container",
+			Value: false,
 		},
 		&cli.PathFlag{
-			Name:    "path",
-			Usage:   "Path to the directory containing the build.envd",
-			Aliases: []string{"p"},
-			Value:   ".",
+			Name:    "private-key",
+			Usage:   "Path to the private key",
+			Aliases: []string{"k"},
+			Value:   sshconfig.GetPrivateKeyOrPanic(),
+			Hidden:  true,
 		},
 		&cli.StringFlag{
-			Name:    "raw",
-			Usage:   "Raw command to execute",
-			Aliases: []string{"r"},
+			Name:  "host",
+			Usage: "Assign the host address for environment ssh acesss server listening",
+			Value: envd.Localhost,
+		},
+		&cli.StringFlag{
+			Name:  "cpu",
+			Usage: "Request CPU resources (number of cores), such as 0.5, 1, 2",
+			Value: "",
+		},
+		&cli.StringFlag{
+			Name:  "memory",
+			Usage: "Request Memory, such as 512M, 2G",
+			Value: "",
+		},
+		&cli.StringFlag{
+			Name:  "gpu",
+			Usage: "Request GPU resources (number of gpus), such as 1, 2",
+			Value: "",
 		},
 	},
-
 	Action: run,
 }
 
 func run(clicontext *cli.Context) error {
-	name := clicontext.String("name")
-	command := clicontext.String("command")
-	rawCommand := clicontext.String("raw")
-	path := clicontext.String("path")
-
-	if command != "" && rawCommand != "" {
-		return errors.New("--raw and --command are mutually exclusive and may only be used once")
+	c, err := home.GetManager().ContextGetCurrent()
+	if err != nil {
+		return errors.Wrap(err, "failed to get current context")
 	}
-
-	resultCommand := rawCommand
-	if command != "" {
-		buildContext, err := filepath.Abs(path)
-		if err != nil {
-			return errors.Wrap(err, "failed to get absolute path of the build context")
-		}
-		fileName, funcName, err := builder.ParseFromStr(clicontext.String("from"))
-		if err != nil {
-			return err
-		}
-		manifest, err := fileutil.FindFileAbsPath(buildContext, fileName)
-		if err != nil {
-			return errors.Wrap(err, "failed to get absolute path of the build file")
-		}
-		if manifest == "" {
-			return errors.Newf("build file %s does not exist", fileName)
-		}
-		opt := builder.Options{
-			ManifestFilePath: manifest,
-			BuildContextDir:  buildContext,
-			BuildFuncName:    funcName,
-		}
-		builder, err := builder.New(clicontext.Context, opt)
-		if err != nil {
-			return errors.Wrap(err, "failed to create the builder")
-		}
-		if err := builder.Interpret(); err != nil {
-			return errors.Wrap(err, "failed to interpret the build file")
-		}
-		if cmd, ok := builder.GetGraph().GetRuntimeCommands()[command]; !ok {
-			return errors.Newf("command %s does not exist", command)
-		} else {
-			resultCommand = cmd
-		}
-		// Get the environment name if `name` is not specified.
-		if name == "" {
-			name = filepath.Base(path)
-		}
+	if c.Runner == types.RunnerTypeDocker {
+		return errors.Newf("docker runner is not supported for this command, please use `envd up`")
 	}
+	telemetry.GetReporter().Telemetry(
+		"run", telemetry.AddField("runner", c.Runner))
```

Comment:
Shall we collect how much time the envd-server takes to create the environment? If so, we should add the duration from the beginning of this function to the detach part.

Reply:
It is not accurate now. For example, I always ctrl c when running run. Then I watch the k8s cluster and ssh when it is ready.